Intersect.Server/Entities/BankInterface.cs

```diff
@@ -319,79 +322,106 @@ public bool TryDepositItem(int slot, int amount, int destSlot = -1, bool sendUpd
                 }
             }
 
-            var itemBase = mPlayer.Items[slot].Descriptor;
-            if (itemBase != null)
+            var inventoryItem = mPlayer.Items[slot];
+            var inventoryItemItemId = inventoryItem.ItemId;
+
+            if (inventoryItem == null || inventoryItemItemId == Guid.Empty)
             {
-                if (mPlayer.Items[slot].ItemId != Guid.Empty)
-                {
-                    if ((!itemBase.CanBank && mGuild == null) || (!itemBase.CanGuildBank && mGuild != null))
-                    {
-                        PacketSender.SendChatMsg(mPlayer, Strings.Items.nobank, ChatMessageType.Bank, CustomColors.Items.Bound);
-                        return false;
-                    }
+                PacketSender.SendChatMsg(mPlayer, Strings.Banks.depositinvalid, ChatMessageType.Bank, CustomColors.Alerts.Error);
+                return false;
+            }
 
-                    lock (mLock)
-                    {
-                        // if stackable, make sure the user actually has this many items or more.
-                        if (itemBase.IsStackable)
-                        {
-                            if (!mPlayer.CanTakeItem(itemBase.Id, amount))
-                            {
-                                amount = mPlayer.FindInventoryItemQuantity(itemBase.Id);
-                            }
-                        }
-                        else
-                        {
-                            amount = 1;
-                        }
+            if ((!inventoryItem.Descriptor.CanBank && mGuild == null) ||
+                (!inventoryItem.Descriptor.CanGuildBank && mGuild != null))
+            {
+                PacketSender.SendChatMsg(mPlayer, Strings.Items.nobank, ChatMessageType.Bank, CustomColors.Items.Bound);
+                return false;
+            }
 
-                        if (CanStoreItem(new Item(itemBase.Id, amount), destSlot))
+            // Calculate the maximum amount that can be deposited, based on quantity and bank space.
+            var depositMultipleItems = mPlayer.FindInventoryItemQuantity(inventoryItemItemId) > 1;
+            var maxAmount = depositMultipleItems ? Math.Min(mPlayer.FindInventoryItemQuantity(inventoryItemItemId),
+                    FindAvailableSpaceForItem(inventoryItemItemId, amount)) : 1;
```

Comment:
`mPlayer.FindInventoryItemQuantity(inventoryItemItemId)` is repeated in both the condition and the branch, the branch is totally unnecessary here.

Reply:
This could be changed to
```csharp
var inventoryQuantity = mPlayer.FindInventoryItemQuantity(inventoryItemItemId);
amount = Math.Min(amount, inventoryQuantity); // This is to optimize the next line
var availableStorageSpace = FindAvailableSpaceForItem(inventoryItemItemId, amount));
var maxAmount = Math.Min(inventoryQuantity, availableStorageSpace);
```